main/cloud_image_validator.py

```diff
@@ -0,0 +1,84 @@
+import json
+import os
+import time
+
+from cloud.terraform.terraform_controller import TerraformController
+from cloud.terraform.terraform_configurator import TerraformConfigurator
+from lib import ssh_lib
+from test_suite.suite_runner import SuiteRunner
+from result.reporter import Reporter
+
+
+class CloudImageValidator:
+    supported_providers = ('aws')
+
+    ssh_identity_file = '/tmp/ssh_key'
+    ssh_pub_key_file = f'{ssh_identity_file}.pub'
+    ssh_config_file = '/tmp/ssh_config'
+
+    infra_controller = None
+    infra_configurator = None
+
+    def __init__(self, resources_file, output_file, parallel, debug):
+        self.resources_file = resources_file
+        self.output_file = output_file
+        self.parallel = parallel
+        self.debug = debug
+
+    def main(self):
+        self.infra_controller = self.initialize_infrastructure()
+
+        try:
+            instances = self.deploy_infrastructure()
+
+            self.run_tests_in_all_instances(instances)
+
+            self.report_test_results()
+        finally:
+            self.cleanup()
+
+    def initialize_infrastructure(self):
```

Comment:
I think we might have to change other modules, but we have functions that are completely private to the class like this one. Should we use the `__function` convention? What is the standard?

Reply:
It's up to us to decide. I don't have a strong opinion on that, unless it's clearly a private method.
E.g. we still don't know if we will reuse these functions for the Jenkins CI (being called from different stages).